pkg/controller/handlers/task/task.go

```diff
@@ -0,0 +1,373 @@
+package task
+
+import (
+	"errors"
+
+	"github.com/obot-platform/nah/pkg/name"
+	"github.com/obot-platform/nah/pkg/randomtoken"
+	"github.com/obot-platform/nah/pkg/router"
+	"github.com/obot-platform/obot/apiclient/types"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	"github.com/obot-platform/obot/pkg/system"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var ErrorValidating = errors.New("found existing external interface, only one external interface can be set")
+
+type Handler struct {
+}
+
+func NewHandler() *Handler {
+	return &Handler{}
+}
+
+func (t *Handler) HandleTaskCreationForCapabilities(req router.Request, _ router.Response) error {
+	thread := req.Object.(*v1.Thread)
+
+	if !thread.Spec.Project {
+		return nil
+	}
+
+	if err := Validate(thread, false); err != nil {
```

Comment:
Why should `forSlack` be false here? What if this task is for slack?

Reply:
The slack one is different. We don't directly set project.capabilities.onSlack, and it is set through controller when slackReceiver is set. So for slack we just have to validate if other configs has been set. So this validate method will onlt be called when interfaces other than slack are set.